src/c++/libos/lwip/lwip_queue.hh

```diff
@@ -66,8 +67,11 @@ class lwip_queue : public io_queue {
     private: static std::map<lwip_4tuple, std::queue<dmtr_sgarray_t> *> our_recv_queues;
     private: static std::unordered_map<std::string, struct in_addr> our_mac_to_ip_table;
     private: static std::unordered_map<in_addr_t, struct rte_ether_addr> our_ip_to_mac_table;
-    private: static uint16_t my_app_port; //FIXME this could/should be a list
-    public: static void set_app_port(uint16_t port) { my_app_port = port; }
+
+    // ???: Should this be static? (all queues share the same my_app_ports instance?)
+    // TODO: Some mechanic for unregistering ports from the application?
```

Comment:
dmtr_close() should

Reply:
That requires looking up the port to which a socket is connected upon `close()`.
Also:
* accepted sockets close() should not unregister a port
* listening sockets close() should unregister the listening port, if and only if there are no active accepted sockets
* connect'ed sockets close() should unregister the connected-to port, if and only if there are no active accepted sockets.

I think that the proper way to do this would be to keep a counter of the number of connections per-port. It is more complicated than just unregistering on close() though